engine/src/main/java/org/destinationsol/systems/DamageSystem.java

```diff
@@ -44,12 +49,14 @@ public EventResult onDamage(DamageEvent event, EntityRef entity) {
         }
         if (entity.getComponent(Health.class).isPresent()) {
             Health health = entity.getComponent(Health.class).get();
-            int newHealthAmount = health.currentHealth - event.getDamage();
-            if (newHealthAmount < 0) {
-                newHealthAmount = 0;
+            health.currentHealth -= event.getDamage();
+            if (health.currentHealth <= 0) {
+                health.currentHealth = 0;
+                entity.setComponent(health);
+                entitySystemManager.sendEvent(new ZeroHealthEvent(), entity);
+            } else {
+                entity.setComponent(health);
```

Comment:
I get this, but can we have a brief comment in the code explaining why `entity.setComponent(health);` has to be here twice please :)